pytorch_lightning/callbacks/batch_size_finder.py

```diff
@@ -0,0 +1,476 @@
+            self._dumped_params["loop_verbose"] = loop.verbose
+
+    def _reset_params(self, trainer):
+        trainer.logger = DummyLogger() if trainer.logger is not None else None
+        trainer.callbacks = []
+
+        if trainer.state.fn == TrainerFn.FITTING:
+            trainer.limit_val_batches = self.steps_per_trial
+            trainer.fit_loop.max_steps = self.steps_per_trial
+        elif trainer.state.fn == TrainerFn.VALIDATING:
+            trainer.limit_val_batches = self.steps_per_trial
+            trainer.validate_loop.verbose = False
+        elif trainer.state.fn == TrainerFn.TESTING:
+            trainer.limit_test_batches = self.steps_per_trial
+            trainer.test_loop.verbose = False
+        elif trainer.state.fn == TrainerFn.PREDICTING:
+            trainer.limit_predict_batches = self.steps_per_trial
+
+    def _restore_params(self, trainer):
+        trainer.logger = self._dumped_params["logger"]
+        trainer.callbacks = self._dumped_params["callbacks"]
+
+        if trainer.state.fn == TrainerFn.FITTING:
+            trainer.fit_loop.global_step = self._dumped_params["global_step"]
+            loop = trainer.fit_loop
+            loop.max_steps = self._dumped_params["max_steps"]
+            trainer.limit_val_batches = self._dumped_params["limit_val_batches"]
+        elif trainer.state.fn == TrainerFn.VALIDATING:
+            loop = trainer.validate_loop
+            trainer.limit_val_batches = self._dumped_params["limit_val_batches"]
+        elif trainer.state.fn == TrainerFn.TESTING:
+            loop = trainer.test_loop
+            trainer.limit_test_batches = self._dumped_params["limit_test_batches"]
+        elif trainer.state.fn == TrainerFn.PREDICTING:
+            loop = trainer.predict_loop
+            trainer.limit_predict_batches = self._dumped_params["limit_predict_batches"]
+
+        loop.load_state_dict(deepcopy(self._dumped_params["loop_state_dict"]), force_load_progress=True)
+        if "loop_verbose" in self._dumped_params:
+            loop.verbose = self._dumped_params["loop_verbose"]
+
+    def pre_early_exit(self, trainer):
+        if trainer.fast_dev_run:
+            return
+
+        # this is required to stop the respective loops
+        if trainer.state.fn == TrainerFn.FITTING:
+            self._dumped_params["num_training_batches"] = trainer.num_training_batches
+            trainer.num_training_batches = 0
+        elif trainer.state.fn == TrainerFn.VALIDATING:
+            self._dumped_params["num_val_batches"] = trainer.num_val_batches
+            trainer.num_val_batches = [0]
+        elif trainer.state.fn == TrainerFn.TESTING:
+            self._dumped_params["num_test_batches"] = trainer.num_test_batches
+            trainer.num_test_batches = [0]
+        elif trainer.state.fn == TrainerFn.PREDICTING:
+            self._dumped_params["num_predict_batches"] = trainer.num_predict_batches
+            trainer.num_predict_batches = [0]
```

Comment:
I believe you should raise a custom expectation and catch it within the Trainer catching error mechanism.